dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java

```diff
@@ -251,7 +250,7 @@ protected void onDestroy() {
         frameworkModel.tryDestroyProtocols();
 
         // 4. destroy application resources
-        for (ModuleModel moduleModel : new ArrayList<>(moduleModels)) {
+        for (ModuleModel moduleModel : moduleModels) {
```

Comment:
Do not change this line. Prevent ConcurrentModificationException here.

Reply:
`moduleModel.destroy` will remove itself in `moduleModels`

Reply:
Hello, I think that moduleModels is CopyOnWriteArrayList, and it returned iterator provides a snapshot of the state of the list, it's not ConcurrentModificationException so we do not to prevent. 
I write a test int ApplicationModelTest.
Please check again